src/character.cpp

```diff
@@ -207,43 +207,49 @@ double Character::aim_per_move( const item& gun, double recoil ) const
         if( mod->sight_dispersion < 0 || mod->aim_cost <= 0 ) {
             continue; // skip gunmods which don't provide a sight
         }
-        if( effective_dispersion( mod->sight_dispersion ) < recoil && mod->aim_cost < cost ) {
-            cost  = mod->aim_cost;
+        if( effective_dispersion( mod->sight_dispersion ) < recoil && mod->aim_cost < sight_cost ) {
+            sight_cost  = mod->aim_cost;
             limit = effective_dispersion( mod->sight_dispersion );
         }
     }
 
-    if( cost == INT_MAX ) {
-        return 0; // no suitable sights (already at maxium aim)
+    if( sight_cost == INT_MAX ) {
+        // No suitable sights (already at maxium aim).
+        return 0;
     }
 
-    // each 5 points (combined) of hand encumbrance increases aim cost by one unit
-    cost += round ( ( encumb( bp_hand_l ) + encumb( bp_hand_r ) ) / 10.0 );
-
-    /** @EFFECT_DEX increases aiming speed */
-    cost += 8 - dex_cur;
+    double aim_speed = 0.0;
 
+    // Ranges [0 - 10]
     /** @EFFECT_PISTOL increases aiming speed for pistols */
     /** @EFFECT_SMG increases aiming speed for SMGs */
     /** @EFFECT_RIFLE increases aiming speed for rifles */
     /** @EFFECT_SHOTGUN increases aiming speed for shotguns */
     /** @EFFECT_LAUNCHER increases aiming speed for launchers */
-    cost += ( ( MAX_SKILL / 2 ) - get_skill_level( gun.gun_skill() ) ) * 2;
+    aim_speed += std::max( MAX_SKILL, static_cast<int>( get_skill_level( gun.gun_skill() ) ) );
```

Comment:
Looks like it should be `min`.

Reply:
Yes it should, thanks, that'll throw the balance back out of whack so I'll need to tweak things a bit.